cpp/CMakeLists.txt

```diff
@@ -783,24 +543,133 @@ set_property(CACHE RMM_LOGGING_LEVEL PROPERTY STRINGS
+        "${CMAKE_SOURCE_DIR}/src/copying"
+        "${CMAKE_SOURCE_DIR}/src/datetime"
+        "${CMAKE_SOURCE_DIR}/src/dictionary"
+        "${CMAKE_SOURCE_DIR}/src/filling"
+        "${CMAKE_SOURCE_DIR}/src/groupby"
+        "${CMAKE_SOURCE_DIR}/src/jit"
+        "${CMAKE_SOURCE_DIR}/src/lists"
+        "${CMAKE_SOURCE_DIR}/src/quantiles"
+        "${CMAKE_SOURCE_DIR}/src/reshape"
+        "${CMAKE_SOURCE_DIR}/src/scalar"
+        "${CMAKE_SOURCE_DIR}/src/search"
+        "${CMAKE_SOURCE_DIR}/src/sort"
+        "${CMAKE_SOURCE_DIR}/src/stream_compaction"
+        "${CMAKE_SOURCE_DIR}/src/strings"
+        "${CMAKE_SOURCE_DIR}/src/structs"
+        "${CMAKE_SOURCE_DIR}/src/table"
+        "${CMAKE_SOURCE_DIR}/src/text"
+        "${CMAKE_SOURCE_DIR}/src/transform"
+        "${CMAKE_SOURCE_DIR}/src/unary"
+        "${CMAKE_SOURCE_DIR}/src/utilities"
+)
+
+set(
+    CUDF_MODULE_LIST
+        "${CMAKE_SOURCE_DIR}/src/ast"
+        "${CMAKE_SOURCE_DIR}/src/comms"
+        "${CMAKE_SOURCE_DIR}/src/hash"
+        "${CMAKE_SOURCE_DIR}/src/interop"
+        "${CMAKE_SOURCE_DIR}/src/io"
+        "${CMAKE_SOURCE_DIR}/src/join"
+        "${CMAKE_SOURCE_DIR}/src/merge"
+        "${CMAKE_SOURCE_DIR}/src/partitioning"
+        "${CMAKE_SOURCE_DIR}/src/reductions"
+        "${CMAKE_SOURCE_DIR}/src/replace"
+        "${CMAKE_SOURCE_DIR}/src/reshape"
+        "${CMAKE_SOURCE_DIR}/src/rolling"
+        "${CMAKE_SOURCE_DIR}/src/transpose"
+)
+
+foreach(CUDF_BASE_MODULE_DIR IN LISTS CUDF_BASE_MODULE_LIST)
+    file(GLOB_RECURSE CUDF_BASE_FILES "${CUDF_BASE_MODULE_DIR}/*")
+    list(APPEND CUDF_BASE_SRC ${CUDF_BASE_FILES})
+endforeach()
+
+add_library("${CUDF_NAMESPACE}_base" ${CUDF_BASE_SRC})
+add_library("${CUDF_NAMESPACE}::base" ALIAS "${CUDF_NAMESPACE}_base")
+add_dependencies("${CUDF_NAMESPACE}_base" stringify_run)
+
+foreach(CUDF_MODULE_DIR IN LISTS CUDF_MODULE_LIST)
+    string(  # replace "/" with ";" to create a list of directory leaves for the module directory
+        REPLACE "/" ";"
+            MODULE_DIR_LEAVES
+            ${CUDF_MODULE_DIR}
+    )
+    list(REVERSE MODULE_DIR_LEAVES)
+    list(GET MODULE_DIR_LEAVES 0 CUDF_MODULE_NAME)  # name the module after the last directory leaf
+    add_library_module(${CUDF_NAMESPACE} ${CUDF_MODULE_NAME} ${CUDF_MODULE_DIR})
+    target_link_libraries("${CUDF_NAMESPACE}_${CUDF_MODULE_NAME}" "${CUDF_NAMESPACE}::base")
+    list(APPEND CUDF_MODULES "${CUDF_NAMESPACE}_${CUDF_MODULE_NAME}")
+endforeach()
+
+foreach(CUDF_HEADERS_DIR IN LISTS CUDF_HEADERS_DIRS)
+    file(GLOB_RECURSE HEADERS "${CUDF_HEADERS_DIR}/*.hpp")
+    list(APPEND CUDF_HEADERS ${HEADERS})
+    file(GLOB_RECURSE HEADERS "${CUDF_HEADERS_DIR}/*.cuh")
+    list(APPEND CUDF_HEADERS ${HEADERS})
+endforeach()
+
+file(WRITE ${CUDF_SRC} "// This file is generated by ${CMAKE_CURRENT_LIST_FILE} \n")
+foreach(CUDF_HEADER IN LISTS CUDF_HEADERS)
+    file(APPEND ${CUDF_SRC} "#include \"${CUDF_HEADER}\" \n")
+endforeach()
+
+add_library(cudf ${CUDF_SRC})
```

Comment:
Can we give a more descriptive name than `CUDF_SRC` here? It's a bit confusing that this is really just a file including all of the headers.

Reply:
Yes. What about something like `CUDF_PROXY_SRC`? Or `CUDF_SHIM_SRC`?

Reply:
Take your pick 😄